components/interop/web3j-service/src/main/kotlin/net/corda/interop/web3j/internal/EthereumConnector.kt

```diff
@@ -0,0 +1,172 @@
+            "eth_syncing" to JsonRpcResponse::class.java,
+            "eth_unsubscribe" to JsonRpcResponse::class.java,
+            "eth_sendRawTransaction" to JsonRpcResponse::class.java,
+            "eth_getBlockByNumber" to NonEip1559Block::class.java,
+        )
+    }
+
+
+    /**
+     * Finds if a json string contains a given key
+     *
+     * @param jsonString The Key string to be found.
+     * @param key The Key string to be found.
+     * @return The matching data class from candidateDataClasses, or null if no match is found.
+     */
+    private fun jsonStringContainsKey(jsonString: String, key: String): Boolean {
+        return try {
+            val jsonNode: JsonNode = objectMapper.readTree(jsonString)
+            jsonNode.has(key)
+        } catch (e: Exception) {
+            // Handle any parsing errors here
+            false
+        }
+    }
+
+
+    /**
+     * Finds the appropriate data class from the candidateDataClasses list that fits the JSON structure.
+     *
+     * @param json The JSON string to be parsed.
+     * @return The matching data class from candidateDataClasses, or null if no match is found.
+     */
+    private fun findDataClassForJson(json: String, method: String): Class<*>? {
+        return if (jsonStringContainsKey(json, "error")) {
+            JsonRpcError::class.java
+        } else {
+            expectedReturnTypes[method]
+        }
+    }
+
+    /**
+     * Returns the useful data from the given input based on its type.
+     *
+     * @param input The input data object to process.
+     * @return The useful data extracted from the input as a string, or an empty string if not applicable.
+     */
+    private fun returnUsefulData(input: Any): ProcessedResponse {
+        when (input) {
+            is JsonRpcError -> {
+                throw EVMErrorException(input)
+            }
+
+            is TransactionResponse -> {
+                return try {
+                    ProcessedResponse(true, input.result?.contractAddress)
+                } catch (e: Exception) {
+                    ProcessedResponse(true, input.result.toString())
+                }
+            }
+
+            is JsonRpcResponse -> return ProcessedResponse(true, input.result)
+            is NonEip1559Block -> return ProcessedResponse(true, input.result)
+        }
+        return ProcessedResponse(false, "")
+    }
+
+
+    /**
+     * Makes an RPC request to the Ethereum node and waits for the response.
+     *
+     * @param rpcUrl The URL of the Ethereum RPC endpoint.
+     * @param method The RPC method to call.
+     * @param params The parameters for the RPC call.
+     * @param waitForResponse Set to true if the function should wait for a response, otherwise false.
+     * @param requests The number of requests made so far (used for recursive calls).
+     * @return A Response object representing the result of the RPC call.
+     */
+    private fun makeRequest(
```

Comment:
nit: I think `performRequest` is clearer as I originally thought this would create the request and not send it.

```suggestion
    private fun performRequest(
```